lib/collection.js

```diff
@@ -0,0 +1,291 @@
+        this.some(function (model, index) {
+            var isFound;
+            if (ctx) {
+                isFound = fn.call(ctx, model, index);
+            } else {
+                isFound = fn(model, index);
+            }
+            if (isFound) {
+                found = model;
+                return true;
+            }
+        });
+        return found;
+    },
+
+    /**
+     * @param {Object} conditions
+     * @returns {Array.<Model>}
+     */
+    where: function (conditions) {
+        return this.filter(this._isMatchConditions.bind(this, conditions));
+    },
+
+    /**
+     * @param {Object} conditions
+     * @returns {?Model}
+     */
+    findWhere: function (conditions) {
+        return this.find(this._isMatchConditions.bind(this, conditions));
+    },
+
+    /**
+     * @param {String} propName
+     * @returns {Array}
+     */
+    pluck: function (propName) {
+        return this.map(function (model) {
+            return model.get(propName);
+        });
+    },
+
+    /**
+     * @param {(Model|Object|Array.<(Model|Object)>)} models
+     * @param {Object} [options]
+     * @param {Number} [options.at]
+     * @returns {Collection}
+     */
+    add: function (models, options) {
+        models = Array.isArray(models) ? models : [models];
+        models.forEach(function (model, index) {
+            var currentOptions;
+            model = this._prepareModel(model);
+            if (!this._isExists(model)) {
+                if (options && options.at !== undefined) {
+                    currentOptions = clone(options);
+                    currentOptions.at = options.at + index;
+                }
+                this._addModel(model, currentOptions);
+            }
+        }, this);
+
+        this._updateLength();
+
+        return this;
+    },
+
+    /**
+     * @param {(Model|Array.<Model>)} models
+     * @returns {Collection}
+     */
+    remove: function (models) {
+        models = Array.isArray(models) ? models : [models];
+        models.forEach(function (model) {
+            if (this._isExists(model)) {
+                this._removeModel(model);
+            }
+        }, this);
+
+        this._updateLength();
+
+        return this;
+    },
+
+    /**
+     * @param {Array.<Models>} models
+     * @returns {Collection}
+     */
+    set: function (models) {
+        var modelsToAdd, existsIds = [],
+            modelsToRemove;
+
+        models = models.map(this._prepareModel, this);
+
+        modelsToAdd = models.filter(function (model) {
+            if (this._isExists(model)) {
+                existsIds.push(model.getId());
+            } else {
+                return true;
+            }
+        }, this);
+
+        modelsToRemove = this._models.filter(function (model) {
+            return existsIds.indexOf(model.getId()) === -1;
+        }, this);
+
+        modelsToRemove.forEach(this._removeModel, this);
+        modelsToAdd.forEach(this._addModel, this);
```

Comment:
The order of models inside the collection will be random after partial remove/ add. It could be replace whole list of models with new one.